docs/apis/streaming/windows.md

```diff
@@ -24,1023 +24,608 @@ specific language governing permissions and limitations
+</div>
+</div>
+
+### WindowFunction with Incremental Aggregation
+
+A `WindowFunction` can be combined with either a `ReduceFunction` or a `FoldFunction`. This allows
+to get the benefit of incremental window computation and also have the additional meta information
+that writing a `WindowFunction` provides.
+
+This is an exampel that shows how incremental aggregation functions can be combined with
+a `WindowFunction`.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+DataStream<Tuple2<String, Long>> input = ...;
+
+// for folding incremental computation
+input
+    .keyBy(<key selector>)
+    .window(<window assigner>)
+    .apply(<initial value>, new MyFoldFunction(), new MyWindowFunction());
+
+// for reducing incremental computation
+input
+    .keyBy(<key selector>)
+    .window(<window assigner>)
+    .apply(new MyReduceFunction(), new MyWindowFunction());
 {% endhighlight %}
-    </td>
-  </tr>
-  <tr>
-    <td><strong>Delta trigger</strong></td>
-    <td>
-      <p>
-        A window is periodically considered for being fired (every 5000 milliseconds in the example).
-        A window is actually fired when the value of the last added element exceeds the value of
-        the first element inserted in the window according to a `DeltaFunction`.
-      </p>
+</div>
+
+<div data-lang="scala" markdown="1">
 {% highlight scala %}
-windowedStream.trigger(DeltaTrigger.of(5000.0, { (old,new) => new - old > 0.01 }))
+val input: DataStream[(String, Long)] = ...
+
+// for folding incremental computation
+input
+    .keyBy(<key selector>)
+    .window(<window assigner>)
+    .apply(<initial value>, new MyFoldFunction(), new MyWindowFunction())
+
+// for reducing incremental computation
+input
+    .keyBy(<key selector>)
+    .window(<window assigner>)
+    .apply(new MyReduceFunction(), new MyWindowFunction())
 {% endhighlight %}
-    </td>
-  </tr>
- </tbody>
-</table>
 </div>
-
 </div>
 
-After the trigger fires, and before the function (e.g., `sum`, `count`) is applied to the window contents, an
-optional `Evictor` removes some elements from the beginning of the window before the remaining elements
-are passed on to the function. Flink comes bundled with a set of evictors You can write your own evictor by
-implementing the `Evictor` interface.
+## Dealing with Late Data
 
-<div class="codetabs" markdown="1">
+When working with event-time windowing it can happen that elements arrive late, i.e the
+watermark that Flink uses to keep track of the progress of event-time is already past the
+end timestamp of a window to which an element belongs. Please
+see [event time](apis/streaming/event_time.html) and especially
+[late elements](pis/streaming/event_time.html#late-elements) for a more thorough discussion of
+how Flink deals with event time.
+
+You can specify for a windowed transformation how it should deal with late elements and how much
```

Comment:
Maybe just "You can specify how a windowed transformation should deal with late elements..."?